continuationsPlugin/src/test/scala/continuations/CallsContinuationResumeWithSuite.scala

```diff
@@ -95,3 +88,96 @@ class CallsContinuationResumeWithSuite extends FunSuite, CompilerFixtures:
 
       assertEquals(CallsContinuationResumeWith.unapply(d), None)
   }
+
+  compilerContext.test("It should return the resume argument if the continuation is reachabl") {
+    implicit givenContext =>
+      val source =
+        """
+          |package continuations
+          |
+          |def mySuspend()(using Suspend): Int =
+          | val x = 3
+          | println("Hi")
+          | Continuation.suspendContinuation[Int] { continuation =>
+          |   continuation.resume(Right(1))
+          | }
+          |""".stripMargin
+
+      checkCompile("pickleQuotes", source) {
+        case (tree, ctx) =>
+          given Context = ctx
+
+          val defDefs = tree.filterSubTrees {
+            case DefDef(name, _, _, _) if name.show == "mySuspend" => true
+            case _ => false
+          }
+
+          val foo = defDefs.head.asInstanceOf[tpd.DefDef]
+
+          assertNoDiff(
+            CallsContinuationResumeWith.unapply(foo).get.show,
+            "Right.apply[Nothing, Int](1)")
+      }
+  }
+
+  compilerContext.test("It should return None is the continuation isn't reachable") {
```

Comment:
hey @pabloconde I can have a look at this scenario as part of the other [story](https://app.clickup.com/t/371z8ck),
unless you have already looked at it